functions/__fzf_extract_var_info.fish

```diff
@@ -6,17 +6,17 @@ function __fzf_extract_var_info --argument-names variable_name set_show_output -
     # $variable_name[
     string match --entire --regex "^\\\$$variable_name(?:: set|\[)" <$set_show_output |
 
-    # Strip the variable name from the variable info, replacing...
+    # Strip the variable name from the scope info, replacing...
     # $variable_name: set in global scope
     # ...with...
     # set in global scope
-    string replace --regex "^\\\$$variable_name: " '' |
+    string replace --regex "^\\\$$variable_name: " "" |
 
     # From the lines of values, keep only the index and value, replacing...
     # $variable_name[1]: length=14 value=|variable_value|
     # ...with...
     # [1] variable_value
-    string replace --regex "^\\\$$variable_name(\[.+\]).+\|(.+)\|\$" '\$1 \$2'
+    string replace --regex "^\\\$$variable_name(\[\d+\]).+?\|(.+)\|\$" '\$1 \$2'
```

Comment:
The bug fix is here: make the `.+` that comes after the first capturing group non-greedy by appending a `?`.

A minor unrelated change is using `\d` to match inside the [ ] to make it more obvious that it's supposed to be index numbers.